### PR TITLE
setValue try also default DateTime constructor format, Date input removes time on setValue

### PR DESCRIPTION
**Pull request by MartinMystikJonas**

### CODE REVIEW

File: src/IPub/FormDateTime/Controls/Date.php [hunk 1]
```diff
@@ -458,6 +458,11 @@ public function setValue($value)
 			$rawValue = $value;
 			$value = Utils\DateTime::createFromFormat($this->getDateFormat(TRUE), $value);
 
+			// Try default format
+			if ($value === FALSE) {
+				$value = Utils\DateTime::createFromFormat(\DateTime::ISO8601, $value);
```
Comment (akadlec) [inline src/IPub/FormDateTime/Controls/Date.php]: Please use `ATOM` instead `ISO8601`, it was wrongly implemented in PHP, so they fixed in `ATOM` constant

**Review by akadlec — CHANGES_REQUESTED** [review]

File: src/IPub/FormDateTime/Controls/Date.php (hunk 1 above)
Comment (MartinMystikJonas) [reply src/IPub/FormDateTime/Controls/Date.php]: done

**Comment by MartinMystikJonas** [comment]
I added tests and support for parsing value from http as string not only as array with date, time elements.

So it will accepts both:
['componentName' => ['date' => '2018-12-18', time => '17:00']]
['componentName' => '2018-12-18 17:00']

This allows you to persist component value as string and pass it back to component which is often required by DataGrids and similar components, which needs to persist state of components in URL.

**Comment by MartinMystikJonas** [comment]
@akadlec Could you please check this PR? Thanks

**Review by akadlec — APPROVED** [review]
Good job 👍

**Pull request merged**